### DISCUSSION TITLE
On commit body formatting

### DISCUSSION THREAD

**Discussion by jbergstroem** [chdsbd/kodiak#773] [Q&A]
I would like to have better control over the commit body formatting. Here is my current workflow:

1. user creates PR as draft
2. add commits
3. (certain ci jobs are run to help the developer with feedback/input)
4. marks ready for review
5. a ci job that validates the pr title and body up to first `---` found against conventional commit guidelines
6. if all jobs that are required passes and the right reviewers approve, kodiak will squash/merge with a commit based on the commit title/body

I'm not sure how, but I'd like to have better control on things but – as most know – git trailers are pretty hard to define and parse; meaning conventional commits won't recognize most patterns. Being able to affect just 
newlines between subject, body and footer for instance would be a great quality of life improvement.

I'm not sure this is the task of kodiak, but since it currently touches this as part of a squash workflow I wanted to see if others have had similar problems / solutions.

**Comment by jbergstroem** [comment]
Just following up here. I am torn whether kodiak should be responsible for the formatting; but seeing how it is managing the squash it is definitely tempting. The other approach I'm considering is having an action / app that replaces kodiak and calling on github to squash/merge. It's not very desirable since kodiak also does other things but I can't think of a better way of getting more control over commit messages at the moment..

**Reply by chdsbd** [reply]
I’m open to supporting mustache-like formatting. Maybe we could have a message.template setting that could be a mustache template for the commit message
